### ISSUE TITLE
'cursorline' causes 'foldtext' to be evaluated too often

### ISSUE THREAD

**Issue by blueyed** [vim/vim#2773]
With `:set cursorline` `&foldtext` is evaluated every time when going to another line (i.e. when the cursor line is redrawn).

Given `t-foldtext.vim`:

```vim
let s:c = 0
function! F()
  let s:c += 1
    return 'foldtext: '.s:c
endfunction

set cursorline

set foldtext=F()
set foldmethod=marker
set foldlevel=0

" fold {{{
"
" }}}
```

When running `vim -Nu NONE t-foldtext.vim --cmd 'source %'` moving around
across lines will update the counter.

This also happens with just `:set cursorline`.

I've noticed this with vim-fugitive, which has a custom foldtext setting, and a
custom plugin that (un)sets cursorline dynamically, because `&cursorline` by
itself causes slowness due to too many redraws.

**Comment by bfredl** [comment]
I think this is essentially the same issue as #2584 (in my testing, entire window gets redrawn, not only the  two lines really necessary)

**Comment by chrisbra** [comment]
so closing as duplicate of #2584

**Comment by blueyed** [comment]
@chrisbra 
Not sure if this is really a duplicate.
It might be similar to #2584 in regard to that `hl_combine_attr` is involved, but is a separate issue by itself.

For what it's worth it also happens with v7.4.2367 (i.e. not related to the bisected commit - if this is really the problem after all).

We can keep this closed as a duplicate, but the fix for #2584 should be verified to fix this issue here then, too.

**Comment by bfredl** [comment]
@blueyed but the main problem with #2584 was never combine attr, rather the unnecessary redraws. And redraw evaluates foldline.
